xarray/coding/strings.py

```diff
@@ -183,7 +183,7 @@ def char_to_bytes(arr):
         # can't make an S0 dtype
         return np.zeros(arr.shape[:-1], dtype=np.bytes_)
 
-    if is_chunked_array(arr):
+    if is_chunked_array(arr) and has_chunkmanager(arr):
```

Comment:
This `is_chunked_array(arr) and has_chunkmanager(arr)` pattern becomes necessary because we are now considering the possibility that `is_chunked_array(arr) == True` but `has_chunkmanager(arr) == False`, whereas previously these were assumed to always be consistent.

Reply:
But doesn't  `has_chunkmanager(arr) == True` imply `is_chunked_array(arr) == True`?

Reply:
@headtr1ck I got a notification saying you commented saying

> But doesn't `has_chunkmanager(arr) == True` imply `is_chunked_array(arr) == True`?

(But I can't find your comment.)

It's a good question though. I _think_ there are some array types that don't define a `.chunks` where you might still want to use other `ChunkManager` methods.

In particular JAX is interesting - it has a top-level [`pmap`](https://jax.readthedocs.io/en/latest/_autosummary/jax.pmap.html) function which applies a function over multiple axes of an array similar to `apply_gufunc`. It distributes computation, but not over `.chunks` (which JAX doesn't define), instead over a global variable [`jax.local_device_count`](https://jax.readthedocs.io/en/latest/_autosummary/jax.local_device_count.html#jax.local_device_count). 

This is why I think we should rename `ChunkManager` to `ComputeManager`.

cc @alxmrs

Reply:
> @headtr1ck I got a notification saying you commented saying
> 
> > But doesn't `has_chunkmanager(arr) == True` imply `is_chunked_array(arr) == True`?
> 
> (But I can't find your comment.)
> 
> It's a good question though. I _think_ there are some array types that don't define a `.chunks` where you might still want to use other `ChunkManager` methods.
> 

I came to the same conclusion, that's why I deleted the comment, sry.

Reply:
No worries! I prefer to leave all my half-baked thoughts in the open and double or triple-post 😅 If you were wondering it then other people will definitely have the same question!

> This is why I think we should rename `ChunkManager` to `ComputeManager`.

I could leave this to a second PR, to isolate the breaking changes.

Reply:
@TomNicholas FYI JAX does now support something a bit like chunking via sharding of `jax.Array`, there's a good summary here: https://jax.readthedocs.io/en/latest/notebooks/Distributed_arrays_and_automatic_parallelization.html
IIUC this is now preferred over `pmap`.